lib/index.js

```diff
@@ -0,0 +1,122 @@
+module.exports = {
+    env: {
+        browser: true,
+        es2021: true,
+        node: true,
+    },
+    extends: ["airbnb-base", "airbnb-typescript/base", "prettier"],
```

Comment:
@oleg-semyonov I'm not sure if it's the time, but postponing this discussion will bring problems later. 

I have reservations against [rule 7.1 from airnbnb ruleset](https://github.com/airbnb/javascript#functions--declarations).

My main argument is that using full `function` declarations is basically better for code readability. If it's a function, then let's call it a `function` (and not a `const`).

I can refer to these articles that generally describe my perspective on it:
- https://davidwalsh.name/i-dont-hate-arrow-functions
- https://hudochenkov.com/posts/prefer-function-declarations/

Reply:
I totally agree with you!
My only issue with this is React function components, which IMO still look better as `const`. For every other case I'm already 100% on function declaration

Reply:
@oleg-semyonov well, for functional components, there's also a different point of view. Please look at https://github.com/facebook/create-react-app/pull/8177

Which makes me believe this is only a habit actually.

Reply:
It is, and it should probably be changed by me :D

Reply:
Added the rule in 9f59e533ee89136939e33948057ce90ce97f9840

Reply:
@e1himself there is one scenario where function declaration does more harm than good: `getServerSideProps` in NextJS.

The generic type used there is quite complex, and splitting it in separate argument and return types would be a pain :(
```ts
export const getServerSideProps: GetServerSideProps<BasePageProps> = async (context) => {
    /* Logic for getting the props */
};
```

The type for `GetServerSideProps` looks like this, and I don't think it would be a good idea to define this generic for each page.
![image](https://user-images.githubusercontent.com/1641218/152321497-0c5dfe98-bfb7-4a42-889e-3cdb4319293d.png)

Reply:
I added an override for files in `/pages` directory in NextJS config, this seems to be the most optimal config